### ISSUE TITLE
False positive LT01 with `COMMENT ON FUNCTION ... IS`

### ISSUE THREAD

**Issue by maciejb** [sqlfluff/sqlfluff#5808]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

sqlfluff detects a false violation of LT01 with Postgres code in the form of
```sql
COMMENT ON FUNCTION x IS 'y';
```

If `fix` is applied, it will be changed to:
```sql
COMMENT ON FUNCTION xIS 'y';
```

### Expected Behaviour

No violation should be found.

### Observed Behaviour

Linting fails LT01

```
L:   1 | P:  22 | LT01 | Unexpected whitespace before 'IS' keyword.
                       | [layout.spacing]
```

### How to reproduce

`echo "COMMENT ON FUNCTION x IS 'y';" | sqlfluff lint --dialect postgres -`

### Dialect

postgres

### Version

3.0.3

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by maciejb** [comment]
Worth noting that this is **not** an issue when the function is qualified with arguments, e.g. `COMMENT ON FUNCTION x(text) IS 'y';`

**Comment by WittierDinosaur** [comment]
That is weird 🤔
